internal/services/eventhub/eventhub_resource_test.go

```diff
@@ -514,6 +527,45 @@ resource "azurerm_eventhub" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomString, data.RandomInteger, data.RandomInteger, enabledString)
 }
 
+func (EventHubResource) captureDescriptionWithDataLake(data acceptance.TestData, enabled bool) string {
+	enabledString := strconv.FormatBool(enabled)
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-eventhub-%d"
+  location = "%s"
+}
+resource "azurerm_eventhub_namespace" "test" {
+  name                = "acctest-EHN%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "Standard"
+}
+resource "azurerm_eventhub" "test" {
+  name                = "acctest-EH%d"
+  namespace_name      = azurerm_eventhub_namespace.test.name
+  resource_group_name = azurerm_resource_group.test.name
+  partition_count     = 2
+  message_retention   = 7
+  capture_description {
+    enabled             = %s
+    encoding            = "Avro"
+    interval_in_seconds = 60
+    size_limit_in_bytes = 10485760
+    skip_empty_archives = true
+    destination {
+      name                  = "EventHubArchive.AzureDataLake"
+      archive_name_format   = "Prod_{EventHub}/{Namespace}\\{PartitionId}_{Year}_{Month}/{Day}/{Hour}/{Minute}/{Second}"
+      datalake_account_name = "exampledatalakeaccount"
```

Comment:
can we provision a real datalake and pass that name in here?

Reply:
I'm afraid of not. datalake gen1 was deprecated by azurerm provider：

### Resource `azurerm_data_lake_store`

Azure Data Lake Storage (Gen1) is deprecated and will be retired on 2024-02-29 - as new Data Lake Storage (Gen1) Accounts can no longer be provisioned - this resource is deprecated and will be removed in v3.0 of the Azure Provider. Support for DataLake Storage (Gen2) is available in the 'azurerm_storage_data_lake_gen2_filesystem' resource and Microsoft's migration documentation can be found here: https://docs.microsoft.com/azure/storage/blobs/data-lake-storage-migrate-gen1-to-gen2.

The capture feature doesn't support data lake gen2

Reply:
is there any ETA for when this will support gen2?

for 1 it means we can not test this, and also given no one can provision new datalakes for this feature should we be adding support for a deprecated resource?

Reply:
Unfortunately, there is no ETA for now. I can try emailing the service team for that info but no guarantee. I understand datalake gen1 is not supported by TF anymore, but we better support it as it's still available in azure portal and in case user wants to use it as the destination in EventHub capture feature?

Reply:
wait, how come new ones can be created via the portal but not terraform?

Reply:
The doc says the resource will be retired on Feb 29, 2024. The new account can still be provision before this date according to this [doc](https://azure.microsoft.com/en-us/updates/action-required-switch-to-azure-data-lake-storage-gen2-by-29-february-2024/). 
![image](https://user-images.githubusercontent.com/92154856/185524487-0e6a01f9-46f2-4e34-a8fb-a87f0da7a4a6.png)
Currently, the account can still be provisioned from Azure portal. It seems @tombuildsstuff removed it according to the [changelog](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/guides/3.0-upgrade-guide.html.markdown#resource-azurerm_data_share_dataset_data_lake_gen1). 

Any suggestions to the next actions? shall we add it back or something?

Reply:
It seems it was removed under the belief we could no longer provision them? or is api support dropped but portal support not?

But again i ask if we should be adding a feature for a deprecated resource? that cannot e2e tests/used with terraform? there doesn't seem to be any issues opened to add gen 1 back so should we be adding something that doesn't seem to have a demand?

Reply:
We can still create the azure data lake gen1 account via Api call using the endpoint mentioned [here](https://docs.microsoft.com/en-us/rest/api/datalakestore/accounts/create?tabs=HTTP).

I do agree your point that it may not be necessary for us to add support for a deprecated resource, however, even we are not seeing any opened issue for now, it doesn't mean we won't receive any in the future given the fact that the datalake gen1 will still be available until 2024. So, I still think we better add it in case user would like to provision it in eventhubs?

Reply:
@xiaxyi what's the use-case we're looking to solve here? Given Gen1 is deprecated, I don't think this is something we should introduce to be honest, particularly given that there's precedent for EOL dates to be brought forward where possible - so there's no guarantee when this would be removed (it could be 2024, 2026, or 3 months from now - which would leave us introducing a breaking change).

Reply:
Thanks @tombuildsstuff for the comments. I totally understand our concerns about introducing a retiring resource here. Although we don't have a user story here, considering it's still a valid resource and will be around for quite a while, what if a user comes to us asking for the it, shall we just tell them to do not use it via Terraform just because it will be retired soon? No worries, it's just my concern, I can remove it if you still think it's not worth adding. :)

Reply:
@xiaxyi - we would need to add back gen1 storage resources, and as no one is asking for them, and they are deprecated i suggest not implementing this feature until it supports gen2. if it does come up and someone asks we can always reopen the PR?

Reply:
Ok, I'll put it on hold then.